src/extensions/ResizableImage.ts

```diff
@@ -18,6 +20,11 @@ export type ResizableImageOptions = ImageOptions & {
    * hostnames are allowed.
    */
   isAllowedImgSrc(src: string | null): boolean;
+
+  /**
+   * Optional React component to pass in as a child component to ResizableImage.
+   */
+  childComponent?: FC<NodeViewProps>;
```

Comment:
I think this type is overly restrictive, since we should accept components that aren't "functional components" (i.e. also support class components). As such, we should use `ComponentType` instead of `FC` (or `ElementType`, which is used for `IconComponent` elsewhere in mui-tiptap for instance, which would end up with an equivalent resultant type as `ComponentType` in this case).

Also should change to PascalCase, per React conventions around component naming (e.g. see examples in MUI here https://mui.com/material-ui/react-autocomplete/#virtualization with `PopperComponent` and `ListBoxComponent`).